src/grafana_agent.py

```diff
@@ -433,6 +471,25 @@ def _update_config(self) -> None:
             # Grafana-agent is not yet available so no need to update config
             return
 
+        # Write TLS files
+        if self.cert.enabled:
+            if not (self.cert.cert and self.cert.key and self.cert.ca):
+                self.status.update_config = WaitingStatus("Waiting for TLS certificate.")
+                self.stop()
+                return
+            self.write_file(self._cert_path, self.cert.cert)
+            self.write_file(self._key_path, self.cert.key)
+            self.write_file(self._ca_path, self.cert.ca)
+        else:
+            try:
+                self.read_file(self._cert_path)
+            except (FileNotFoundError, PathError):
+                pass
+            else:
+                self.delete_file(self._cert_path)
+                self.delete_file(self._key_path)
+                self.delete_file(self._ca_path)
```

Comment:
The `try` only has `_cert_path` so these two could still technically raise.
But why do this at all? Why not `recursive=True` and drop the `read_file`?

Reply:
there is no `recursive=True` for `delete_file` but I addressed it by checking for each file individually here.